build/build.go

```diff
@@ -1612,3 +1619,40 @@ func noPrintFunc(opt map[string]Options) bool {
 	}
 	return true
 }
+
+func saveLocalState(so client.SolveOpt, opt Options, node builder.Node, configDir string) error {
+	var err error
+
+	if so.Ref == "" {
+		return nil
+	}
+
+	lp := opt.Inputs.ContextPath
+	dp := opt.Inputs.DockerfilePath
+	if lp != "" || dp != "" {
+		if lp != "" {
+			lp, err = filepath.Abs(lp)
```

Comment:
:thinking: This doesn't seem right, think we need something more similar to: https://github.com/docker/buildx/blob/b716e48926298a5be3f2381a6fb09d6cd442272f/commands/build.go#L683

Reply:
Ignore me, I hadn't realized this was on `SolveOpt` - so it's definitely a path to a local file at this point.